app/src/main/java/fr/free/nrw/commons/achievements/AchievementsActivity.java

```diff
@@ -186,23 +197,42 @@ void shareScreen(Bitmap bitmap) {
      */
     private void setAchievements() {
         if (checkAccount()) {
-            compositeDisposable.add(mediaWikiApi
-                    .getAchievements(Objects.requireNonNull(sessionManager.getCurrentAccount()).name)
-                    .subscribeOn(Schedulers.io())
-                    .observeOn(AndroidSchedulers.mainThread())
-                    .subscribe(
-                            response -> {
-                                if (response != null) {
-                                    setUploadCount(Achievements.from(response));
-                                } else {
+            //Timber.d(String.valueOf(mediaWikiApi.getAchievements(sessionManager.getCurrentAccount().name)));
+                /*layoutImageReverts.setVisibility(View.VISIBLE);
+                imageView.setVisibility(View.VISIBLE);*/
+            try{
+
+                compositeDisposable.add(mediaWikiApi
+                        .getAchievements(Objects.requireNonNull(sessionManager.getCurrentAccount()).name)
+                        .subscribeOn(Schedulers.io())
+                        .observeOn(AndroidSchedulers.mainThread())
+                        .subscribe(
+                                response -> {
+                                    if (response != null) {
+                                        setUploadCount(Achievements.from(response));
+                                    } else {
+                                        Timber.d("success");
+                                        layoutImageReverts.setVisibility(View.INVISIBLE);
+                                        imageView.setVisibility(View.INVISIBLE);
+                                        levelNumber.setText("You haven't qualified any level yet");
+                                        //onError();
+                                    }
+                                },
+                                t -> {
+                                    Timber.e(t, "Fetching achievements statistics failed");
                                     onError();
                                 }
-                            },
-                            t -> {
-                                Timber.e(t, "Fetching achievements statistics failed");
-                                onError();
-                            }
-                    ));
+                        ));
+            }
+            catch (Exception e){
+                Timber.d(e+"success");
+                    /*layoutImageReverts.setVisibility(View.INVISIBLE);
```

Comment:
Please clear this comments, they are not descriptive.

Reply:
Fixed in the next commit